Ex1/main.java

```diff
@@ -49,18 +51,186 @@ public static void main(String[] args) {
+                    //loadFromFile(phone_book);
                     break;
                 case "11":
                     System.out.println("Goodbye !");
+                    flag = false;
                     break;
             }
         }
     }
+
+    public static void addContact(ArrayList<Contact> phone_book)
+    {
+        Scanner sForAdd = new Scanner (System.in); //open a new scanner for input
+        /*a do-while loop for taking care of invalid phone numbers, using .next() for only
+        taking strings with no backspace but check the phoneNumber with !matches() if the user is trying to "be funny"
+        the "\\d+" is a regular expression that says "the string contain only digits."
+        if matches() true (the string contains only digits) -> !matches() is false -> skip the error and carry on with the script
+        if matches() false (the string don't contain only digits) -> !matches() is true -> show error to user and go back.
+         */
+        String phoneNumber;
+        String fullName;
+        do
+        {
+            System.out.println("Enter the contact phone number -->");
+            phoneNumber = sForAdd.next();
+            if(!phoneNumber.matches("\\d+")) {
+                System.out.println("The phone number you have entered is invalid (Empty/only white Space String) - Try again");
+            }
+        }while(!phoneNumber.matches("\\d+"));
+        /*a do-while loop for taking care of invalid names, using .nextLine() for
+        taking care of strings but check the fullName with isBlank() if the user is trying to "be funny"
+        if isBlank() true -> fullName is invalid -> show the error and repeat the loop
+        if isBlank() false -> fullName is valid -> carry on with the script.
+         */
+        sForAdd.nextLine(); //clearing the buffer of the scanner.
+        do
+        {
+            System.out.println("Enter the contact full name -->");
+            fullName = sForAdd.nextLine();
+            if(fullName.isBlank()) {
+                System.out.println("The name you have entered is invalid (Empty/only white Space String) - Try again");
+            }
+        }while(fullName.isBlank());
+
+        //here we enter the values to a placeHolder of type Contact and pushing it into the linked list.
+        Contact placeHolder = new Contact(fullName,phoneNumber);
+        phone_book.add(placeHolder);
+    }
+
+    public static void printContacts(ArrayList<Contact> phone_book)
+    {
+        // function number 3 - prints the Contacts in the phone book
+        // if the phone book is empty
+        if (phone_book.isEmpty()) {
+            System.out.println("The Phone Book is empty!");
+        }
+        else
+        {
+            System.out.println("Phone Book Contacts:\n----------------------");
+            // make an iterator & pass all over the phone book
+            Iterator<Contact> iterator = phone_book.iterator();
+            while (iterator.hasNext())
```

Comment:
```suggestion
            for(Contact con : phone_book)
```
instead of using iterators you can use this simple for loop

Reply:
This is Shir's function.